### PR TITLE
feat: update clientPayment status

### PR DESCRIPTION
**Pull request by lissavxo**
Related to #1021 

Depends on
---
- [ ] #1034 

Description
---
Update Client Payment register when transaction is made

Test plan
---
Create a payment Id with the endpoint api/payments/paymentId?address=your_address, make a payment with the payment Id created in the op_return, make sure the payment Id register status updates according to transaction status

## Summary by CodeRabbit

- **New Features**
  - Add payment-ID creation API with CORS, input validation, and standardized success/error responses; backend now records client payments with statuses and exposes generate/get/update payment APIs; blockchain processing updates client-payment statuses on mempool/confirmation events.

- **Bug Fixes**
  - Standardize "Method Not Allowed" responses to HTTP 405 and improve mapping of address-related errors to proper responses.

- **Tests**
  - Add extensive unit tests covering payment-status update logic and edge cases.

### CODE REVIEW

File: services/chronikService.ts [hunk 1]
```diff
@@ -445,17 +457,18 @@ export class ChronikBlockchainClient {
         }
         this.mempoolTxsBeingProcessed += 1
         console.log(`${this.CHRONIK_MSG_PREFIX}: [${msg.msgType}] ${msg.txid}`)
-        const transaction = await this.chronik.tx(msg.txid)
```
[the remark this answers is not in the document]
Comment (lissavxo) [reply services/chronikService.ts:448]: this was actually moved to before the ifs chain, so I can use in the TX_CONFIRMED and TX_ADDED_TO_MEMPOOL msgType conditions

**Review by chedieck — APPROVED** [review]
